### ISSUE TITLE
Contacting you for this project - Please read

### ISSUE THREAD

**Issue by EternalModz** [Atten007/Minecraft-Console-Updater#3]
Hey, I know that this doesn't really count as an 'issue' but I wanted to get in touch with you. This project is quite dificult and I understand what you are trying to do. I'd like to talk to you and this was one of my only methods of contacting you. I'm not sure what the best form of contacting you is but my Discord ID is: EternalModz#7322 but yeah, thanks for reading this and yeah, I hope we can get in touch soon!

**Comment by Korozin** [comment]
A bit late, so I apologize. But I wouldn't hold your breath on this. From what I've seen, I highly doubt this person actually knows how to even go about _starting_ this project. If they wanted to actually add the newer updates to existing Legacy editions, they would first off need extremely comprehensive reverse-engineering knowledge of the Wii U architecture itself, C++, and actual PPC Assembly- and even after learning that it would take years to fully disassemble the game to a point where it can be readily modified and run without issues on a scale like that.

What would have a better yield is creating their _own_ version of Minecraft from the ground up, and compiling it to a WUP Package for the Wii U to install and run. This would remove a lot of the reverse-engineering aspects, but would leave you needing to either A: Use Nintendo's actual dev package, which iirc they don't even issue anymore, and it would be illegal to release to other people, or B: circle back to needing ASM knowledge to compile your own project.

In short? This project probably isn't ever gonna happen, at least not from the owner of this repo.

**Comment by EternalModz** [comment]
Makes sense. Actually, I have discovered an easier method for doing it on the PS3 using an SPRX with custom hooks. Additionally, my friends and I have successfully implemented custom content into the game by utilizing this method, as opposed to fully reversing the game and instead, using a custom C++ executable. I think it’d be pretty cool if we could somehow find a way to do something similar on other legacy consoles. I can confirm that this works on the PS3, PS4 and PS5. But I cannot say anything about other consoles. But yeah,  it’d be epic if this was possible on other consoles. Also, there are teams that are working on reversing the game, I know many that started a few years ago and their still working on it today. But this is quite interesting and I honestly hope it works out.

**Comment by Korozin** [comment]
The easiest way to do it in my opinion (apart from custom-building your own engine), would be utilizing code-caves and custom Client writes. Since I'm (or used to be) a Wii U modder, going to use the general thought process as the example.

The main Game Executable or the .RPX client can be edited if you take certain steps, which I'm not going to go into full detail... But if you add the base features you need, you can then use a Memory Hook utilizing Code Caves (or empty spaces in the RAM) to call on these features without directly interfering with normal Game or Memory processes. But it is a _very_ lengthy and unstable process. For stability's sake, making your own engine would not only be easier in an editing and deploying aspect, but it would be SO MUCH easier to work with your own code, rather than Executable Assembly, or Raw RAM Data.

**Comment by EternalModz** [comment]
Yeah, true.

**Comment by EternalModz** [comment]
> The easiest way to do it in my opinion (apart from custom-building your own engine), would be utilizing code-caves and custom Client writes. Since I'm (or used to be) a Wii U modder, going to use the general thought process as the example.
> 
> The main Game Executable or the .RPX client can be edited if you take certain steps, which I'm not going to go into full detail... But if you add the base features you need, you can then use a Memory Hook utilizing Code Caves (or empty spaces in the RAM) to call on these features without directly interfering with normal Game or Memory processes. But it is a _very_ lengthy and unstable procedure. For stability's sake, making your own engine would not only be easier in an editing and deploying aspect, but it would be SO MUCH BETTER to work with your own code, rather than Executable Assembly, or Raw RAM / Hex Data.

Hey, do you have Discord? I’d like to talk to you.

**Comment by Korozin** [comment]
Sadly, no. The only socials I have are listed on my profile.

**Comment by EternalModz** [comment]
Ah, ok. Would you be able to get it? I’m wanting to talk about a project I’m working on with my team. It’s for Minecraft. It’ll be a world converter + NBT editor + a bunch more. I would like to talk to you about this because it’d be great if you joined the team. Here’s the link to the project: https://github.com/The-Minecraft-Manipulator-Team/Minecraft-Manipulator

**Comment by Korozin** [comment]
I'm not even able to get Discord since the domain itself has been completely wiped from the DNS Registry- and I'm also unable to change it since there's a DNS Forwarder set up to force configure the blocking onto every device- and I refuse to hack the network to get around that.

Anyways, I wouldn't be all that useful on this project anyways since my only experience with Legacy Editions is from the Wii U version, on top of that I don't even own the console anymore since my last one broke so I would have nothing to reference. The only language I'm good at is Python as well, and since I'm assuming you're going the .NET C# Route, I wouldn't be able to contribute much.

Anyways, most of the goals that I can see from your Project Page could be pretty easily ripped from UME and essentially pasted into a new GUI. Don't see why you'd need me for that.

**Comment by maxfire2008** [comment]
> add the newer updates to existing Legacy editions
I *think* that what it means is porting the Bedrock Edition to legacy consoles. At least that's what "you’ve to buy Minecraft: Bedrock Edition in the Store" seems to suggest.

**Comment by EternalModz** [comment]
Makes sense.

**Comment by Korozin** [comment]
> I _think_ that what it means is porting the Bedrock Edition to legacy consoles. At least that's what "you’ve to buy Minecraft: Bedrock Edition in the Store" seems to suggest.

That idea would be even _more_ unrealistic then as it would require a full de-compilation of Bedrock. Which with only one person would take literal years as this wouldn't require just the game. It would also require the imported SDKs which might also be proprietary, along with other dependencies.

If that goal were ever to be even remotely realistic- it would require a full-team of extremely experienced Devs with a LOT of time to kill.
